OpenRA.Mods.AS/Projectiles/CyclicImpactProjectile.cs

```diff
@@ -0,0 +1,262 @@
+
+		[Desc("Delay in ticks until trail animation is spawned.")]
+		public readonly int TrailDelay = 1;
+
+		[Desc("Palette used to render the trail sequence.")]
+		[PaletteReference("TrailUsePlayerPalette")]
+		public readonly string TrailPalette = "effect";
+
+		[Desc("Use the Player Palette to render the trail sequence.")]
+		public readonly bool TrailUsePlayerPalette = false;
+
+		public readonly int ContrailLength = 0;
+		public readonly int ContrailZOffset = 2047;
+		public readonly Color ContrailColor = Color.White;
+		public readonly bool ContrailUsePlayerColor = false;
+		public readonly int ContrailDelay = 1;
+		public readonly WDist ContrailWidth = new WDist(64);
+
+		[Desc("Is this blocked by actors with BlocksProjectiles trait.")]
+		public readonly bool Blockable = true;
+
+		[Desc("Width of projectile (used for finding blocking actors).")]
+		public readonly WDist Width = new WDist(1);
+
+		[Desc("If projectile touches an actor with one of these stances during or after the first bounce, trigger explosion.")]
+		public readonly Stance ValidBounceBlockerStances = Stance.Enemy | Stance.Neutral | Stance.Ally;
+
+		public IProjectile Create(ProjectileArgs args) { return new CyclicImpactProjectile(this, args); }
+
+		void IRulesetLoaded<WeaponInfo>.RulesetLoaded(Ruleset rules, WeaponInfo info)
+		{
+			WeaponInfo weapon;
+			if (!rules.Weapons.TryGetValue(Weapon.ToLowerInvariant(), out weapon))
+				throw new YamlException("Weapons Ruleset does not contain an entry '{0}'".F(Weapon.ToLowerInvariant()));
+			WeaponInfo = weapon;
+		}
+	}
+
+	public class CyclicImpactProjectile : IProjectile, ISync
+	{
+		readonly CyclicImpactProjectileInfo info;
+		readonly ProjectileArgs args;
+		[Sync]
+		readonly WDist speed;
+
+		[Sync]
+		WPos projectilepos, targetpos, sourcepos, offsetTargetPos = WPos.Zero;
+		WPos offsetSourcePos = WPos.Zero;
+		int lifespan;
+		int ticks;
+		int mindelay;
+		World world;
+		WRot offsetRotation;
+		CyclicImpactProjectileEffect[] projectiles; // offset projectiles
+
+		public Actor SourceActor { get { return args.SourceActor; } }
+
+		public CyclicImpactProjectile(CyclicImpactProjectileInfo info, ProjectileArgs args)
+		{
+			this.info = info;
+			this.args = args;
+
+			projectilepos = args.Source;
+			sourcepos = args.Source;
+
+			var firedBy = args.SourceActor;
+
+			world = args.SourceActor.World;
+
+			if (info.Speed.Length > 1)
+				speed = new WDist(world.SharedRandom.Next(info.Speed[0].Length, info.Speed[1].Length));
+			else
+				speed = info.Speed[0];
+
+			targetpos = GetTargetPos();
+
+			mindelay = args.Weapon.MinRange.Length / speed.Length;
+
+			projectiles = new CyclicImpactProjectileEffect[info.Offsets.Count()];
+			var range = Util.ApplyPercentageModifiers(args.Weapon.Range.Length, args.RangeModifiers);
+			var mainfacing = (targetpos - sourcepos).Yaw.Facing;
+
+			// target that will be assigned
+			Target target;
+
+			// main bullet facing
+			int facing = 0;
```

Comment:
The comment here is confusing. Go with projectileFacing IMO and reword the comment to refer that this is used for the subprojs.